File: python/questions/minRemoval.py
from typing import List

class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        l = 0
        maxl = 0
        for r in range(0, len(nums)):
            while nums[l]*k < nums[r]:
                l += 1
            maxl = max(maxl, r-l+1)
        return len(nums) - maxl

File: cpp/questions/minRemoval.cpp
#include <vector>

using namesapce std;

class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        std::sort(nums.begin(), nums.end());
        int l = 0;
        int maxl = 0;
        for (int r=0; r<nums.size(); r++) {
            while (nums[l]*1L*k < nums[r]) {
                l++;
            }
            maxl = std::max(maxl, r-l+1);
        }
        return nums.size() - maxl;
    }
};
